internal/server/data/data.go

```diff
@@ -122,9 +122,8 @@ func (d *DB) GormDB() *gorm.DB {
 	return d.DB
 }
 
-// TODO: accept sql.TxOptions when we remove gorm
-func (d *DB) Begin(ctx context.Context) (*Transaction, error) {
-	tx := d.DB.WithContext(ctx).Begin()
+func (d *DB) Begin(ctx context.Context, opts *sql.TxOptions) (*Transaction, error) {
```

Comment:
could consider using variadic arguments to allow the opts to be optional.
```suggestion
func (d *DB) Begin(ctx context.Context, opts ...sql.TxOptions) (*Transaction, error) {
```

Reply:
I see `gorm` uses this pattern, but I consider this dangerous and I don't think it should ever be used.

varargs are great if you are going to use all the args, but using varargs for a single optional argument and ignoring the rest is going to cause confusion for anyone reading the code. Adding a `nil` for the parameter isn't really a big inconvenience. We could also follow the stdlib and have two functions for `Begin`, but at this time it didn't seem necessary since it was easy enough to add the nils.

Reply:
Well, it's just a suggestion. The nil argument is a bit ugly, especially when that's the default case. From the project I've seen, this is a really common pattern and familiar to many Go devs.